lib/elixir/lib/calendar/date.ex

```diff
@@ -942,7 +942,7 @@ defmodule Date do
         %Date{calendar: calendar, year: year, month: month, day: day}
 
       {day_of_week, first_day_of_week, _} ->
-        add(date, -(day_of_week - first_day_of_week))
+        add(date, -Integer.mod(day_of_week - first_day_of_week, 7))
```

Comment:
This can be replaced with 
```elixir
add(date, -Integer.mod(day_of_week - first_day_of_week, max(first_day_of_week, last_day_of_week))
```

Reply:
Are you sure this would work if the first day is a saturday?

Reply:
I'm doing some more testing and adding support for non-1-indexed ranges as well. I'll update the PR when I've done more testing on all possible variations of starting days.

Reply:
You're right José, there's no concrete way to calculate the days in a week from a start_day and end_day pair.